projects/batfish/src/test/java/org/batfish/bdp/BdpDataPlanePluginTest.java

```diff
@@ -181,6 +205,30 @@ public void testIbgpRejectOwnAs() throws IOException {
   }
 
   @Test
+  public void testIbgpRejectSameNeighborID() throws IOException {
+    String testrigName = "ibgp-reject-routerid-match";
+    String[] configurationNames = new String[] {"r1", "r2", "r3", "r4"};
+    Batfish batfish =
+        BatfishTestUtils.getBatfishFromTestrigResource(
+            TESTRIGS_PREFIX + testrigName, configurationNames, null, null, null, null, _folder);
+    BdpDataPlanePlugin dataPlanePlugin = new BdpDataPlanePlugin();
+    dataPlanePlugin.initialize(batfish);
+    dataPlanePlugin.computeDataPlane(false);
+    SortedMap<String, SortedMap<String, SortedSet<AbstractRoute>>> routes =
+        dataPlanePlugin.getRoutes();
+    SortedSet<AbstractRoute> r2Routes = routes.get("r2").get(Configuration.DEFAULT_VRF_NAME);
+    SortedSet<AbstractRoute> r3Routes = routes.get("r3").get(Configuration.DEFAULT_VRF_NAME);
+    Set<Prefix> r2Prefixes = r2Routes.stream().map(r -> r.getNetwork()).collect(Collectors.toSet());
+    Set<Prefix> r3Prefixes = r3Routes.stream().map(r -> r.getNetwork()).collect(Collectors.toSet());
```

Comment:
Style advice: I like to structure my tests into 2/3 clear sections.

1. Initialization ("nothing interesting here") and constants.
2. The actual code that is under test. As short as possible.
3. Assertions. (perhaps merged with 2, if 2 is one line.)

I usually have whitespace and/or comments between them.

Also, make use of helpers for any common code. (see `makeFlow`, `makeAcl`, etc.)

An example: https://github.com/batfish/batfish/commit/fd022f5d46fae6670758245fa747e5e707d59d85#diff-7ee7600d68323ad23eae06bf4434516cR71

I learned this over time as a pattern that makes tests readable -- all the "background" is at the top and all the "tests" are at the bottom. Whitespace helps.

Feel free to go your own way.